LBH.AdultSocialCare.Api/V1/Gateways/NursingCarePackageGateway.cs

```diff
@@ -61,6 +64,32 @@ public async Task<NursingCarePackage> UpsertAsync(NursingCarePackage nursingCare
                 : null;
         }
 
+        public async Task<NursingCarePackage> ChangeStatusAsync(Guid nursingCarePackageId, int statusId)
+        {
+            NursingCarePackage nursingCarePackageToUpdate = await _databaseContext.NursingCarePackage
+                .Include(item => item.Clients)
+                .Include(item => item.Status)
+                .Include(item => item.NursingCareAdditionalNeeds)
+                .FirstOrDefaultAsync(item => item.Id == nursingCarePackageId)
+                .ConfigureAwait(false);
+
+            if (nursingCarePackageToUpdate == null)
+            {
+                throw new ErrorException($"Couldn't find the record: {nursingCarePackageId}");
+            }
+            nursingCarePackageToUpdate.StatusId = statusId;
+            await _databaseContext.SaveChangesAsync().ConfigureAwait(false);
+            return nursingCarePackageToUpdate;
```

Comment:
Save can fail. Wrap save async in try catch and in catch throw `DBSaveFailedException` with an error saying save or update failed. Error handler will take care of it.

Reply:
Make sense!